hledger-lib/Hledger/Reports/AccountTransactionsReport.hs

```diff
@@ -94,63 +95,71 @@ triCommodityBalance c = filterMixedAmountByCommodity c  . triBalance
 accountTransactionsReport :: ReportSpec -> Journal -> Query -> AccountTransactionsReport
 accountTransactionsReport rspec@ReportSpec{_rsReportOpts=ropts} j thisacctq = items
   where
-    -- a depth limit should not affect the account transactions report
-    -- seems unnecessary for some reason XXX
-    reportq = simplifyQuery $ And [_rsQuery rspec, periodq, excludeforecastq (forecast_ ropts)]
+    -- a depth limit should not affect the account transactions report, nor
+    -- should cur or amt queries, as they are handled separately
+    reportq = simplifyQuery $ And [depthlessq, periodq, excludeforecastq (forecast_ ropts)]
       where
+        depthlessq = filterQuery (not . queryIsSymOrAmt) . filterQuery (not . queryIsDepth) $ _rsQuery rspec
         periodq = Date . periodAsDateSpan $ period_ ropts
         -- Except in forecast mode, exclude future/forecast transactions.
         excludeforecastq (Just _) = Any
         excludeforecastq Nothing  =  -- not:date:tomorrow- not:tag:generated-transaction
           And [ Not . Date $ DateSpan (Just . addDays 1 $ _rsDay rspec) Nothing
               , Not generatedTransactionTag
               ]
-    symq    = filterQuery queryIsSym reportq
-    realq   = filterQuery queryIsReal reportq
-    statusq = filterQuery queryIsStatus reportq
+    symq = filterQuery queryIsSymOrAmt $ _rsQuery rspec
+    queryIsSymOrAmt q = queryIsSym q || queryIsAmt q
 
-    -- sort by the transaction's register date, for accurate starting balance
-    -- these are not yet filtered by tdate, we want to search them all for priorps
-    transactions =
-        ptraceAtWith 5 (("ts4:\n"++).pshowTransactions)
-      . sortOn (transactionRegisterDate reportq thisacctq)
-      . jtxns
-      . ptraceAtWith 5 (("ts3:\n"++).pshowTransactions.jtxns)
-      -- maybe convert these transactions to cost or value
-      . journalApplyValuationFromOpts rspec
-      . (if filtertxns then filterJournalTransactions reportq else id)
-      -- keep just the transactions affecting this account (via possibly realness or status-filtered postings)
-      . traceAt 3 ("thisacctq: "++show thisacctq)
-      . ptraceAtWith 5 (("ts2:\n"++).pshowTransactions.jtxns)
-      . filterJournalTransactions thisacctq
-      . filterJournalPostings (And [realq, statusq])
-      . ptraceAtWith 5 (("ts1:\n"++).pshowTransactions.jtxns)
-      -- apply any cur:SYM filters in reportq
-      $ if queryIsNull symq then j else filterJournalAmounts symq j
+    -- Note that within this functions, we are only allowed limited
+    -- transformation of the transaction postings: this is due to the need to
+    -- pass the original transactions into accountTransactionsReportItem.
+    -- Generally, we either include a transaction in full, or not at all.
+    -- We only allow ourselves to:
+    --   - filter amounts (e.g. cur: or amt:)
+    --   - apply valuation or costing
+    --
```

Comment:
I wasn't sure if this meant filtering postings within transactions or whole transactions. How about:
```
  - filter transactions by amt: and/or cur:
  - apply valuation or costing to transaction amounts
```
I'm still a little confused, why can we filter by cur:/amt: here but not desc: or status: ?

Reply:
These were chose based on going back to hledger-1.20 and documenting what was done there, rather than a systematic consideration of what would be appropriate. I could update the comment to give the rationale and list these as examples.

Reply:
Maybe this would be enough ?
```
-- Do some limited filtering and valuing of the journal's transactions:
-- filter them by the account query if any,
-- discard amounts not matched by the currency query if any,
-- then apply valuation if any.
```